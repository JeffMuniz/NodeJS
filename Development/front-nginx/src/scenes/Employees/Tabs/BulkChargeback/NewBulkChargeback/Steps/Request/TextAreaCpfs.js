import React, { PureComponent } from "react";
import { func, string, bool, oneOfType } from "prop-types";

import { TextArea } from "./TextAreaCpfs.styled";

export class TextAreaCpfs extends PureComponent {
  inputParser = e => {
    const { key } = e;
    const keyboard = RegExp(/^[0-9-.\b ]+$/);

    if (key === "Enter") {
      return true;
    }

    if (!keyboard.test(key)) {
      e.preventDefault();
      return false;
    }
  };

  render() {
    const { name, onBlur, hasError, onChange } = this.props;

    return (
      <TextArea
        name={name}
        id="textarea-cpfs"
        placeholder="Ex.:&#10;000.000.000-00&#10;000.000.000-00&#10;000.000.000-00"
        onKeyPress={this.inputParser}
        onBlur={onBlur}
        onChange={onChange}
        hasError={hasError}
      />
    );
  }
}

TextAreaCpfs.propTypes = {
  onChange: func.isRequired,
  name: string.isRequired,
  onBlur: func.isRequired,
  hasError: oneOfType([string, bool]),
};

TextAreaCpfs.defaultProps = {
  hasError: false,
};

export default TextAreaCpfs;
