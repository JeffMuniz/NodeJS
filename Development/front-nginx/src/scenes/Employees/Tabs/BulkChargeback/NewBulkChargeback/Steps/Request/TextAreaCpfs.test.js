import React from "react";
import TestRenderer from "react-test-renderer";

import { TextAreaCpfs } from "./TextAreaCpfs";

const props = {
  onChange: jest.fn(),
  name: "teste",
  onBlur: jest.fn(),
};

describe("TextAreaCpfs", () => {
  it("Render Text Area", () => {
    const testRenderer = TestRenderer.create(<TextAreaCpfs {...props} />);
    const testInstance = testRenderer.root;
    const formInstance = testInstance.findByType(TextAreaCpfs).instance;

    expect(
      formInstance.inputParser({ key: "a", preventDefault: jest.fn() }),
    ).toBe(false);

    expect(
      formInstance.inputParser({ key: "Enter", preventDefault: jest.fn() }),
    ).toBe(true);

    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Render Text Area Error ", () => {
    const testRenderer = TestRenderer.create(
      <TextAreaCpfs hasError {...props} />,
    );

    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
