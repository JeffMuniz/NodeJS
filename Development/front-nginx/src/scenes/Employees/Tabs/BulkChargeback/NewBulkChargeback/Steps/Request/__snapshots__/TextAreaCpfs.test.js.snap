// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaCpfs Render Text Area 1`] = `
<textarea
  className="sc-bdVaJa jtTXiQ"
  id="textarea-cpfs"
  name="teste"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onKeyPress={[Function]}
  placeholder="Ex.:
000.000.000-00
000.000.000-00
000.000.000-00"
/>
`;

exports[`TextAreaCpfs Render Text Area Error  1`] = `
<textarea
  className="sc-bdVaJa cmPVGY"
  id="textarea-cpfs"
  name="teste"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  onKeyPress={[Function]}
  placeholder="Ex.:
000.000.000-00
000.000.000-00
000.000.000-00"
/>
`;
